### PR TITLE
354 web download apk from the website

### LINKED ISSUE #354: [WEB] Download APK from the website
- [ ] Avoir un texte "Télécharger" dans le Header.
- [ ] Le texte est cliquable et redirige vers la route "/download".
- [ ] La route "/download" permet de récupérer l'APK généré dans le container de l'application mobile.
- [ ] La route "/download" doit renvoyer 200 si il n'y a pas d'erreur sinon 40X.
- [ ] Tests unitaires.

### PR DESCRIPTION
**Pull request by H-chauvet**
Allow the user to download the mobile application from the website, by clicking on "Download the application" in the "Header" of the website.

Close #354

### CODE REVIEW

**Review by 31Nathan — CHANGES_REQUESTED** [review]
Missing doc for the route

**Review by 31Nathan — APPROVED** [review]

File: Application/mobile/Dockerfile [hunk 1]
```diff
@@ -10,5 +10,14 @@ RUN rm -f .packages
...
-RUN mkdir /apk/
-ENTRYPOINT cp build/app/outputs/apk/release/app-release.apk /apk/client.apk
+# Debug: List contents of the build directory
+RUN ls -l build/app/outputs/flutter-apk/
+
+RUN mkdir -p /apk
+RUN cp build/app/outputs/flutter-apk/app-release.apk /apk/client.apk
+
+# Debug: List contents of the apk directory
+RUN ls -l /apk/
```
Comment (CedricCORGE) [inline Application/mobile/Dockerfile:20]: if it's debug, we don't need it anymore

File: Application/mobile/Dockerfile [hunk 2]
```diff
@@ -10,5 +10,14 @@ RUN rm -f .packages
 RUN flutter pub get
 RUN flutter build apk
 
-RUN mkdir /apk/
-ENTRYPOINT cp build/app/outputs/apk/release/app-release.apk /apk/client.apk
+# Debug: List contents of the build directory
+RUN ls -l build/app/outputs/flutter-apk/
```
Comment (CedricCORGE) [inline Application/mobile/Dockerfile:14]: if it's debug, we don't need it anymore

File: Configurateur/front/lib/components/custom_header.dart
Comment (CedricCORGE) [inline Configurateur/front/lib/components/custom_header.dart:1]: we didn't say that it's would be better to do the download on the showcase website ?

**Review by CedricCORGE — CHANGES_REQUESTED** [review]

File: Configurateur/front/lib/components/custom_header.dart
Comment (H-chauvet) [reply Configurateur/front/lib/components/custom_header.dart:1]: Showcase website is not accessible to the public, risu website is

**Review by CedricCORGE — APPROVED** [review]

**Pull request merged**